### PR TITLE
Adds test code

### PR DESCRIPTION
**Pull request by jmjkim**
내일 너가 얘기한 test부분만 따로 분리해서? 하는게 나을것같다고 한 부분 설명해주면 다음부터 그렇게 하도록할께!

### CODE REVIEW

**Comment by morishjs** [comment]
@jmjkim 
File Changes 를 보면 Test 코드와 관련없이 줄바꿈만 된 파일이 있지? (src/components/TextField.tsx) 
보통 PR을 올릴 때는 PR 주제와 관련된 것들만 올리는게 convention 이고 매너이고 나랑 할 때 최대한 그걸 체화시키면 좋을 것 같아

File: test/App.test.tsx [hunk 1]
```diff
@@ -5,40 +5,41 @@
...
-        expect(passwordInput);
-    });
-
-    test('Input values for Name and Password should be validated(min(5), max(10))', () => {
-        userEvent.type(nameInput, 'abc');
+describe('', () => {
+    it('render error messages if validation fails', () => {
+        render(
+            <SimpleForm>
+                <TextField
```
Comment (morishjs) [inline test/App.test.tsx]: 너가 짠 코드도 맞는데 테스트 코드 자체가 다른 사람이 코드를 읽을 때 의도가 분명하게 드러나게 좋거든 그래서 우리가 구현한 TextField, SelectField, CheckboxField 각각 정상 케이스, 잘못된 케이스(Edge case) 각각 만들면 좋을 것 같아

**Review by morishjs — CHANGES_REQUESTED** [review]

File: test/App.test.tsx (hunk 1 above)
Comment (morishjs) [reply test/App.test.tsx]: 그럼 총 6개가 되겠지?

Comment (morishjs) [reply test/App.test.tsx]: 6개는 it('...') 으로 구분해주면 돼

**Comment by jmjkim** [comment]
이따가 미팅때 질문들 할게 많네.. PR convention은 오늘 집가서 확인해볼게

**Comment by jmjkim** [comment]
준석아 <TextField>나 <SelectboxField>에는 에러 메세지가 원하는대로 발생했는데 <CheckboxField>에는 에러 메세지를 발생시키기 위해 두번 클릭 (입력값을 없앰으로써 에러 메세지 발생)을 해도 에러 메세지가 안나오는데 혹시 어떤식으로 문제에 접근해야하는지 아이디어좀 줄수있겠니?

**Comment by morishjs** [comment]
@jmjkim 

```javascript
        const {queryAllByRole, getByText} = render(
            <SimpleForm>
                <CheckboxField
                    type={'checkbox'}
                    source={'location'}
                    label={'거주 국가'}
                    validate={[required()]}
                />
            </SimpleForm>
        );
```

SimpleForm 으로 감싸줘야 해 왜냐하면 SimpleForm 에서 상태관리를 하거든 (error, value 같은)

**Comment by jmjkim** [comment]
omg..ㅋㅋㅋthat was super easy..got it thanks!!

**Comment by jmjkim** [comment]
시간되면 리뷰 부탁할께!

File: test/App.test.tsx [hunk 2]
```diff
@@ -2,43 +2,93 @@
...
+                    source={'name'}
+                    label={'이름'}
+                    validate={[min(5), max(10)]}
+                />
+            </SimpleForm>
+        );
+        const name = getByLabelText('이름');
+
+        it('#1 min(5) - render an error message if input value is less than 5.', () => {
```
Comment (morishjs) [inline test/App.test.tsx:31]: 강제는 아니지만 it should... 으로 시작하는게 통상적으로 사용되는 룰이라서 

`it should render an error message... #1 min(5)`
이렇게 하는게 맞을 것 같아 근데 사소한 것이라서 approve 할게

**Review by morishjs — APPROVED** [review]

File: test/App.test.tsx (hunk 2 above)
Comment (jmjkim) [reply test/App.test.tsx:31]: 아 오케이 참고할께! 너말대로 통상적인 룰을 따르는게 좋을것같은데 수정해서 다시 PR해도될까?

Comment (morishjs) [reply test/App.test.tsx:31]: @jmjkim 
물론이지
